src/pds/ingress/util/backoff_util.py

```diff
@@ -0,0 +1,29 @@
+"""
+===============
+backoff_util.py
+===============
+
+Module containing functions related to utilization of the backoff module for
+automatic backoff/retry of HTTP requests.
+
+"""
+import requests
+
+
+def fatal_code(err: requests.exceptions.RequestException) -> bool:
+    """
+    Determines if the HTTP return code associated with a requests exception
+    corresponds to a fatal error or not. If the error is of a transient nature,
+    this function will return False, indicating to the backoff decorator that
+    the reqeust should be retried. Otherwise, a return value of True will
+    cause any backoff/reties to be abandoned.
+    """
+    if err.response is not None:
+        # HTTP codes indicating a transient error (including throttling) which
+        # are worth retrying after a backoff
+        transient_codes = [408, 425, 429, 500, 502, 503, 504, 509]
```

Comment:
I wonder if it'd be helpful to use the constants in [http.HTTPStatus](https://docs.python.org/3/library/http.html) ~as although I have memorized every one of the HTTP codes~ okay I have no idea what some of these codes are! 425 is "too early"? What‽

Reply:
Nice suggestion, I have added use of the enum everywhere I was using magic numbers for HTTP codes